mypy/checker.py

```diff
@@ -508,7 +502,6 @@ def check_func_item(self, defn: FuncItem,
         if isinstance(defn, FuncDef):
             fdef = defn
 
-        self.function_stack.append(defn)
```

Comment:
Because the signature is checked in the scope of the class. I have moved the push to line 645, just before checking the body. Is it wrong?